src/tls/25519.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "pem.h"
+#include "tls.h"
+
+namespace tls
+{
+  // This function parses x25519 PEM keys generated by openssl (e.g. for
+  // members' public encryption key) and returns the raw 32-byte key.
+  static std::vector<uint8_t> parse_25519_public(const Pem& public_pem)
```

Comment:
I'm surprised we can't do this with the existing `tls::PublicKey`, I think `mbedtls_pk_write_pubkey_der` would write these raw key bytes?

Reply:
I thought so too. However, we'd need to load the PEM into a `mbedtls_pk_context` first and this does not work as `25519` keys are not support by the version of TLS we use.

Reply:
I have added a comment to mention this limitation at the top of the file.

Reply:
Ah of course, fair enough!